core/capabilities/ccip/common/extradatacodec.go

```diff
@@ -11,11 +17,47 @@ func NewExtraDataCodec() ExtraDataCodec {
 }
 
 func (c ExtraDataCodec) DecodeExtraArgs(extraArgs cciptypes.Bytes, sourceChainSelector cciptypes.ChainSelector) (map[string]any, error) {
-	// Not implemented and not return error
-	return nil, nil
+	if len(extraArgs) == 0 {
+		// return empty map if extraArgs is empty
+		return nil, nil
+	}
+
+	family, err := chainsel.GetSelectorFamily(uint64(sourceChainSelector))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get chain family for selector %d: %w", sourceChainSelector, err)
+	}
+
+	switch family {
+	case chainsel.FamilyEVM:
+		return ccipevm.DecodeExtraArgsToMap(extraArgs)
+
+	case chainsel.FamilySolana:
+		return ccipsolana.DecodeExtraArgsToMap(extraArgs)
+
+	default:
+		return nil, fmt.Errorf("unsupported family for extra args type %s", family)
+	}
 }
 
 func (c ExtraDataCodec) DecodeTokenAmountDestExecData(destExecData cciptypes.Bytes, sourceChainSelector cciptypes.ChainSelector) (map[string]any, error) {
```

Comment:
Why do we need this additional wrapper? Why not use ccipevm/ccipsolana's functions directly in the plugin definition here? https://github.com/smartcontractkit/chainlink/blob/20c3b69f3cb7d2941475151f345ae4058fd1fd01/core/capabilities/ccip/oraclecreator/plugin.go#L255

Keeping this switch case in common just means more places we need to update when adding a new chain family

Reply:
Thanks for pointing this out! totally agree, I made a PR for it https://github.com/smartcontractkit/chainlink/pull/16223